server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecification.java

```diff
@@ -151,11 +156,15 @@ public void addSystemDependencies(final Collection<ModuleDependency> dependencie
     public void addUserDependency(final ModuleDependency dependency) {
         allDependencies = null;
         this.userDependencies.add(dependency);
+        this.logImplicitDependency(dependency.getIdentifier().getName(), dependency.getIdentifier().getSlot(), dependency.getReason());
```

Comment:
User dependencies are dependencies that the user has specifically added, either via jboss-deployment-structure.xml or via the manifest, so they are not implicit ones, and I believe they will end up without having a reason, it depends on how we want to code them.

Those are dependencies added by the user, so I wouldn't add a reason for them and consequently, I wouldn't try to log them.